### PR TITLE
parser: Fix NO_WRITE_TO_BINLOG option for ALTER TABLE Partition Operations syntax

### PR DESCRIPTION
**Pull request by spongedu**
### What problem does this PR solve? 
For #445, support `NO_WRITE_TO_BINLOG` option for partition related operations in `ALTER TABLE` syntax

### What is changed and how it works?
1. Add `NoWriteToBinLogAliasOpt` in related parser item.
2. Add a flag in AST node `AlterTableSpec` and fix it's `Restore` method
3. Add related tests

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test
 - No code

Code changes

 N / A

Side effects

N / A

Related changes

N / A

### CODE REVIEW

**Comment by spongedu** [comment]
@leoppro  PTAL

File: parser_test.go [hunk 1]
```diff
@@ -2074,6 +2074,8 @@ func (s *testParserSuite) TestDDL(c *C) {
 				PARTITION P2 VALUES LESS THAN (2015),
 				PARTITION P3 VALUES LESS THAN MAXVALUE)`, true, "ALTER TABLE `employees` ADD PARTITION (PARTITION `P1` VALUES LESS THAN (2010), PARTITION `P2` VALUES LESS THAN (2015), PARTITION `P3` VALUES LESS THAN (MAXVALUE))"},
 		{"alter table t add partition (partition x values in ((3, 4), (5, 6)))", true, "ALTER TABLE `t` ADD PARTITION (PARTITION `x` VALUES IN ((3, 4), (5, 6)))"},
+		{"ALTER TABLE employees ADD PARTITION NO_WRITE_TO_BINLOG", true, "ALTER TABLE `employees` ADD PARTITION NO_WRITE_TO_BINLOG"},
+		{"ALTER TABLE employees ADD PARTITION NO_WRITE_TO_BINLOG PARTITIONS 10", true, "ALTER TABLE `employees` ADD PARTITION NO_WRITE_TO_BINLOG PARTITIONS 10"},
```
Comment (leoppro) [inline parser_test.go]: add test for `local`

**Comment by leoppro** [comment]
rest LGTM

**Comment by leoppro** [comment]
@tangenta PTAL

File: parser_test.go (hunk 1 above)
Comment (spongedu) [reply parser_test.go]: ok

Comment (spongedu) [reply parser_test.go]: @leoppro  PTAL :P

**Comment by leoppro** [comment]
LGTM

**Pull request merged**